runtime/syntax/qml.vim

```diff
@@ -0,0 +1,1145 @@
+" Vim syntax file
+" Language:     QML
+" Maintainer:   Peter Hoeg <peter@hoeg.com>
+" Updaters:     Refer to CONTRIBUTORS.md
+" URL:          https://github.com/peterhoeg/vim-qml
+" Changes:      `git log` is your friend
+" Last Change:  2017-11-11
+"
+" This file is bassed on the original work done by Warwick Allison
+" <warwick.allison@nokia.com> whose did about 99% of the work here.
+
+" Based on javascript syntax (as is QML)
+
+if !exists("main_syntax")
+  if version < 600
+    syntax clear
+  elseif exists("b:current_syntax")
+    finish
+  endif
+  let main_syntax = 'qml'
+endif
```

Comment:
The version check has been removed from the distributed syntax files as Vim 6 was released over twenty years ago.

This can be simplified to:

```vim
  if exists("b:current_syntax")
    finish
  endif

  if !exists("main_syntax")
    let main_syntax = 'qml'
  endif
```